### ISSUE TITLE
Scripts and binaries are not executable

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#22718]
On the latest v8 beta release, executables like `./gradlew`, `kubectl-apply-sh` etc are not executable and I had to do `chmod +x` on all of them. Can someone confirm if its just me or the file templates are indeed not executable

**Comment by mshima** [comment]
Looks like release package is missing executable bit (maven too):
```
ls -la node_modules/generator-jhipster/dist/generators/gradle/templates
total 32
drwxr-xr-x  5 mshima  staff   160 26 Jun 09:56 .
drwxr-xr-x  9 mshima  staff   288 26 Jun 09:56 ..
drwxr-xr-x  3 mshima  staff    96 26 Jun 09:56 gradle
-rw-r--r--  1 mshima  staff  8496 26 Jun 09:56 gradlew
-rw-r--r--  1 mshima  staff  2868 26 Jun 09:56 gradlew.bat
```

**Comment by jelharou** [comment]
I actually have the same issue and need to chmod +x both `./gradlew` and `./npmw`

**Comment by deepu105** [comment]
its not just gradle and maven, all binaries and script files have the issue. So might be an issue with git config when someone checked them in. Btw i'm on Linux not sure if the behavior is different on a mac

**Comment by jelharou** [comment]
I'm unable to reproduce the issue with a forked copy of main.
(I'm on ChromeBook/Debian)

**Comment by mshima** [comment]
There is no issue with source code. The issue is with released package.
Releases should be created with posix computers (don’t know if that is the case) otherwise the executable bit will be missing.
Can only be fixed by whom have release rights.

**Comment by DanielFran** [comment]
Not sure what is needed in git config...
I did the release using windows os.

**Comment by mraible** [comment]
Maybe we should publish the release using GitHub Actions and build it with Ubuntu? 

I use a [publish action](https://github.com/oktadev/schematics/blob/main/.github/workflows/publish.yml) for OktaDev Schematics. It publishes to npmjs.org when a release is created.

**Comment by mraible** [comment]
Increased bug bounty to help get this fixed. Maybe you can do the release, @deepu105?

**Comment by mshima** [comment]
@DanielFran already prepared a new release for this weekend.

**Comment by deepu105** [comment]
Ah Windows issue again. Are you sure it's not the git filemode setup that
is downloading the binaries as non executable on windows? Else there be
options to setup file mods in NPM so that they are preserved. I did a quick
search but didn't find any. But I would assume it's a common problem with
solutions

**Comment by jelharou** [comment]
@DanielFran The git filemode is configured correctly but you may want to double check locally :
```
jelharou@penguin:~/jelharou/generator-jhipster$ more .git/config 
[core]
        repositoryformatversion = 0
        filemode = true
```

Apart from deploying from a non windows machine, we could add a postinstall script to the package.lock but that will be cumbersome as there are many files and they depend on the individual project configuration.

@mraible suggestion to use a publish action may be ideal.

**Comment by DanielFran** [comment]
@deepu105 @mshima @jelharou Can you confirm if with beta.2 the issue is fixed?

**Comment by jelharou** [comment]
Yes, npmw and gradlew now have the correct executable permissions on linux.

**Comment by mraible** [comment]
On macOS, it seems like `npmw` and `mvnw` don't have the correct permissions.

<img width="1682" alt="Screenshot 2023-07-13 at 12 45 18 AM" src="https://github.com/jhipster/generator-jhipster/assets/17892/28e94088-c226-47a9-ab07-5f0c1c772cf6">

**Comment by mshima** [comment]
Executable bit still missing:
```
% cat /usr/local/lib/node_modules/generator-jhipster/package.json | grep version
  "version": "8.0.0-beta.2",
```

```
% ls -la /usr/local/lib/node_modules/generator-jhipster/dist/generators/maven/templates
total 40
drwxr-xr-x   5 mshima  wheel    160 13 Jul 10:30 .
drwxr-xr-x  10 mshima  wheel    320 13 Jul 10:30 ..
drwxr-xr-x   4 mshima  wheel    128 13 Jul 10:30 .mvn
-rw-r--r--   1 mshima  wheel  11289 13 Jul 10:30 mvnw
-rw-r--r--   1 mshima  wheel   7796 13 Jul 10:30 mvnw.cmd
```

```
% ls -la /usr/local/lib/node_modules/generator-jhipster/dist/generators/gradle/templates
total 32
drwxr-xr-x  5 mshima  wheel   160 13 Jul 10:30 .
drwxr-xr-x  9 mshima  wheel   288 13 Jul 10:30 ..
drwxr-xr-x  3 mshima  wheel    96 13 Jul 10:30 gradle
-rw-r--r--  1 mshima  wheel  8496 13 Jul 10:30 gradlew
-rw-r--r--  1 mshima  wheel  2868 13 Jul 10:30 gradlew.bat
```

**Comment by mshima** [comment]
Missing on linux devcontainer too:

```sh
vscode ➜ /workspaces/generator-jhipster (main) $ npm install generator-jhipster@latest

added 22 packages, removed 41 packages, changed 108 packages, and audited 1212 packages in 1m

273 packages are looking for funding
  run `npm fund` for details

4 moderate severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
vscode ➜ /workspaces/generator-jhipster (main) $ cat node_modules/generator-jhipster/package.json | grep version
  "version": "8.0.0-beta.2",
    "latest-version": "7.0.0",
vscode ➜ /workspaces/generator-jhipster (main) $ ls -la node_modules/generator-jhipster/dist/generators/maven/templates/
total 20
drwxr-xr-x  5 vscode vscode   160 Jul 13 13:38 .
drwxr-xr-x 10 vscode vscode   320 Jul 13 13:38 ..
drwxr-xr-x  4 vscode vscode   128 Jul 13 13:38 .mvn
-rw-r--r--  1 vscode vscode 11289 Jul 13 13:38 mvnw
-rw-r--r--  1 vscode vscode  7796 Jul 13 13:38 mvnw.cmd

```

**Comment by DanielFran** [comment]
@mraible @mshima The fix I did was to add core.filemode=true in git config file

It is missing any other parameter?

**Comment by mshima** [comment]
Are you releasing with windows? Or windows + wsl?
I don't think it's possible releasing with pure windows, not sure.
You can check the dist folder permission:

```
npm run clean
npm run build
ls -la dist/generators/maven/templates
```

**Comment by DanielFran** [comment]
@mshima Pure windows.
Will try tonight...

**Comment by mshima** [comment]
I suppose this was fixed by v8.0.0-beta.3
